apps/lib/apps.c

```diff
@@ -406,7 +406,7 @@ CONF *app_load_config_internal(const char *filename, int quiet)
     BIO *in = NULL; /* leads to empty config in case filename == "" */
     CONF *conf;
 
-    if (*filename != '\0'
+    if (filename != NULL && *filename != '\0'
```

Comment:
To solve the @paulidale concern it should be as simple as doing:
```
if ((filename != NULL && *filename != '\0')
    || (in = bio_open_default_(filename, 'r', FORMAT_TEXT, quiet)) == NULL)

Reply:
Do not read from stdin for config. It will likely cause unexpected behavior for any other command (dgst, enc, etc) that reads from stdin.

Reply:
@richsalz This just reverts the behavior to 1.1.1 version.

Reply:
> @richsalz This just reverts the behavior to 1.1.1 version.

Doesn't mean it isn't/wasn't wrong.

Reply:
I do not think it is wrong. Also not doing it would also break the backwards compat for the spkac command.

Reply:
> I do not think it is wrong. Also not doing it would also break the backwards compat for the spkac command

Your vote counts more than mine. :)  And this isn't just spkac, it's for all commands unless I read the diff excerpt wrong.

But hey, do what you think best.

Reply:
@t8m what would be the conclusion on this?

Reply:
What I proposed just gets the implementation back to the 1.1.1 state of things. So please use that. If Rich has any concerns with that he can open a separate issue.